### PR TITLE
Bugfix/linux task tracking

### PR DESCRIPTION
**Pull request by adrianherrera**
This fixes an important bug with `LinuxMonitor::getPid`.

The current implementation of `getPid` uses the TSS and `current_thread_info` objects to get a pointer to the current `task_struct`, from which we can read the PID from. Unfortunately, this method is no longer valid for kernel versions > 4.x

This required a lot of refactoring in the S2E Linux kernel. However, it now removes the hardcoded struct offsets from the plugin and makes it usable across multiple architectures and kernel versions.

### CODE REVIEW

File: src/s2e/Plugins/OSMonitors/Linux/BaseLinuxMonitor.h [hunk 1]
```diff
@@ -26,20 +26,22 @@ extern CPUX86State *env;
...
 
-// from arch/x86/include/asm/page_32_types.h
-#define THREAD_SIZE_ORDER 1
-#define THREAD_SIZE (PAGE_SIZE << THREAD_SIZE_ORDER)
+///
+/// \brief The start address of the kernel
+///
+/// This is configured in the Linux kernel via the CONFIG_PAGE_OFFSET option. This is the default value for X86.
+///
+static const uint64_t KERNEL_START_ADDRESS = 0xC0000000;
```
Comment (vitalych) [inline src/s2e/Plugins/OSMonitors/Linux/BaseLinuxMonitor.h]: Can we pass this from the driver as well?

File: src/s2e/Plugins/OSMonitors/Linux/DecreeMonitor.cpp [hunk 2]
```diff
@@ -234,7 +243,7 @@ klee::ref<klee::Expr> DecreeMonitor::readMemory8(S2EExecutionState *state, uint6
 uint64_t DecreeMonitor::getPid(S2EExecutionState *state, uint64_t pc) {
     target_ulong pid;
     target_ulong taskStructPtr = getTaskStructPtr(state);
-    unsigned pidAddress = taskStructPtr + 320; // Offsets can be found with: pahole vmlinux -C task_struct
+    target_ulong pidAddress = taskStructPtr + decree::TASK_STRUCT_PID_OFFSET;
```
Comment (vitalych) [inline src/s2e/Plugins/OSMonitors/Linux/DecreeMonitor.cpp]: Pass this through the kernel driver as well

File: src/s2e/Plugins/OSMonitors/Linux/BaseLinuxMonitor.h (hunk 1 above)
Comment (adrianherrera) [inline src/s2e/Plugins/OSMonitors/Linux/BaseLinuxMonitor.h]: Sure.

File: src/s2e/Plugins/OSMonitors/Linux/DecreeMonitor.cpp (hunk 2 above)
Comment (adrianherrera) [inline src/s2e/Plugins/OSMonitors/Linux/DecreeMonitor.cpp]: Ok. Currently I only added the kernel module to the vanilla Linux kernel. I'll have to add it to the Decree version to do this.

**Comment by adrianherrera** [comment]
Ok, I've ported the kernel module to the Decree kernel, so this removed a few more hard-coded constants.

Ready to be reviewed again!

**Pull request merged**